### PR TITLE
1 model input preprocessing

### LINKED ISSUE #1: Enhancement: Improve Preprocessing of Mixed Data Types in Model Input
Our model currently lacks a preprocessing step for input data. This enhancement aims to introduce a preprocessing pipeline to handle both categorical and numerical data types, improving model performance and data consistency.

Develop a preprocessing pipeline that includes:
-Feature scaling for numerical data to normalize distributions.
-One-hot encoding for categorical data to enable better model processing.
-Ensure the pipeline integrates seamlessly with the current model input.

### PR DESCRIPTION
**Pull request by Keithan-Smith**
Complete the preprocessing of categorical and numerical variables in the input data. This is necessary to achieve appropriate training and testing data for the implementation of a supervised model in the next step, and for the fine-tuning of the supervised models.

### CODE REVIEW

File: src/consumer_loans.ipynb [hunk 1]
```diff
@@ -16,7 +16,8 @@
     "  - [Numerical variable distribution](#Numerical-variable-distribution)\n",
     "  - [Boxplots](#Boxplots)\n",
     "- [Preprocessing the data](#Preprocessing-the-data)\n",
```
Comment (blaise0577) [inline src/consumer_loans.ipynb]: I guess it's not an issue, but the indentation of line 18 is off by a space.

File: src/consumer_loans.ipynb [hunk 2]
```diff
@@ -100,107 +101,107 @@
        "  <tbody>\n",
```
Comment (blaise0577) [inline src/consumer_loans.ipynb]: If this is to create a table in markdown, there is another way using hyphens and pipes for splitting rows and columns.  Can be easier to read when in pure markdown.  But maybe this html is auto-generated within github.

File: src/consumer_loans.ipynb [hunk 3]
```diff
@@ -507,17 +508,17 @@
...
-       "std      6.545265e+04\n",
+       "mean     6.625363e+04\n",
+       "std      6.495657e+04\n",
        "min      4.000000e+03\n",
-       "25%      3.840000e+04\n",
-       "50%      5.500000e+04\n",
-       "75%      7.885100e+04\n",
+       "25%      3.900000e+04\n",
+       "50%      5.520000e+04\n",
+       "75%      8.000000e+04\n",
```
Comment (blaise0577) [inline src/consumer_loans.ipynb]: Just from a reviewer's perspective, it's hard to know what this piece of json is about.  Could these be extracted into separate files and then the filename would explain what the data is?  If not, no worries.

File: src/consumer_loans.ipynb [hunk 4]
```diff
@@ -936,14 +937,214 @@
     "# Initialize the OneHotEncoder\n",
     "encoder = OneHotEncoder(sparse=False, drop='first')\n",
     "\n",
-    "# Fit and transform the categorical columns\n",
-    "encoded_data = encoder.fit_transform(df_train[catvars])\n",
+    "# Fit the encoder on the categorical columns of the training data and then use that fitted encoder to transform test data\n",
+    "# Ensure consistency in how categorical variables are handled between training and test sets\n",
+    "encoder.fit(df_train[catvars])\n",
```
Comment (blaise0577) [inline src/consumer_loans.ipynb]: Is this line still needed?  Seems to do something but not care about the output.

File: src/consumer_loans.ipynb [hunk 5]
```diff
@@ -936,14 +937,214 @@
...
+    "\n",
+    "encoded_data_train = encoder.transform(df_train[catvars])\n",
```
Comment (blaise0577) [inline src/consumer_loans.ipynb]: Feels like you can create a method as the same three lines of code happens to the train and test data, and then trigger the method twice.

File: src/consumer_loans.ipynb [hunk 6]
```diff
@@ -936,14 +937,214 @@
...
+    "### 3.2 Preprocessing Numerical Variables"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 42,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df_train_targets = df_train['loan_status']\n",
```
Comment (blaise0577) [inline src/consumer_loans.ipynb]: Again, a function would reduce duplication here

File: src/consumer_loans.ipynb [hunk 7]
```diff
@@ -936,14 +937,214 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 45,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.preprocessing import StandardScaler, MinMaxScaler\n",
+    "# Instantiate models\n",
+    "standard_scaler = StandardScaler()\n",
+    "# minmax_scaler = MinMaxScaler()\n",
```
Comment (blaise0577) [inline src/consumer_loans.ipynb]: Remove commented code

File: src/consumer_loans.ipynb [hunk 8]
```diff
@@ -936,14 +937,214 @@
...
+    "\n",
+    "# Fit and Transform the training numerical variables using StandardScaler\n",
```
Comment (blaise0577) [inline src/consumer_loans.ipynb]: Do these comments aid the understanding of the code?

File: src/consumer_loans.ipynb (hunk 1 above)
Comment (Keithan-Smith) [reply src/consumer_loans.ipynb:18]: done

File: src/consumer_loans.ipynb (hunk 3 above)
Comment (Keithan-Smith) [reply src/consumer_loans.ipynb]: interesting this is actually the output from one of the code chunks in the ipynb file
Capture

File: src/consumer_loans.ipynb (hunk 4 above)
Comment (Keithan-Smith) [reply src/consumer_loans.ipynb]: yes this line is needed to fit the encoder onto the categorical variable columns wihtin the training data

File: src/consumer_loans.ipynb (hunk 5 above)
Comment (Keithan-Smith) [reply src/consumer_loans.ipynb]: done, bear in mind architecture of the project will change to reflect proper ML project styling

**Comment by Keithan-Smith** [comment]
merged
